pgjdbc/src/test/java/org/postgresql/test/jdbc2/BinaryTest.java

```diff
@@ -0,0 +1,282 @@
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * TestCase to test handling of binary types.
+ */
+public class BinaryTest {
+  private Connection con;
+
+  /**
+   * Set up the fixture for this testcase: the tables for this test.
+   *
+   * @throws SQLException if a database error occurs
+   */
+  @Before
+  public void setUp() throws SQLException {
+    con = TestUtil.openDB();
+
+    TestUtil.createTable(con, "test_binary", "id integer,name text,geom point");
+
+    TestUtil.closeDB(con);
+  }
+
+  /**
+   * Tear down the fixture for this test case.
+   *
+   * @throws SQLException if a database error occurs
+   */
+  @After
+  public void tearDown() throws SQLException {
+    TestUtil.closeDB(con);
+
+    con = TestUtil.openDB();
+
+    TestUtil.dropTable(con, "test_binary");
+
+    TestUtil.closeDB(con);
+  }
+
+  /**
+   * Make sure the functions for adding binary transfer OIDs for custom types are correct.
+   *
+   * @throws SQLException if a database error occurs
+   */
+  @Test
+  public void testBinaryTransferOids() throws SQLException {
+    con = TestUtil.openDB();
+    QueryExecutor queryExecutor = ((PgConnection) con).getQueryExecutor();
+    // get current OIDs (make a copy of them)
+    Set<Integer> oidsReceive = new HashSet<Integer>(queryExecutor.getBinaryReceiveOids());
+    Set<Integer> oidsSend = new HashSet<Integer>(queryExecutor.getBinarySendOids());
+    // add a new OID to be transferred as binary data
+    int customTypeOid = 91716;
+    // first for receiving
+    queryExecutor.addBinaryReceiveOid(91716);
+    // check new OID
+    assertTrue(queryExecutor.useBinaryForReceive(customTypeOid));
+    assertFalse(queryExecutor.useBinaryForSend(customTypeOid));
+    // make sure all other OIDs are still there
+    for (int oid : oidsReceive) {
+      assertTrue(queryExecutor.useBinaryForReceive(oid));
+    }
+    for (int oid : oidsSend) {
+      assertTrue(queryExecutor.useBinaryForSend(oid));
+    }
+    // then for sending
+    queryExecutor.addBinarySendOid(91716);
+    // check new OID
+    assertTrue(queryExecutor.useBinaryForReceive(customTypeOid));
+    assertTrue(queryExecutor.useBinaryForSend(customTypeOid));
+    // make sure all other OIDs are still there
+    for (int oid : oidsReceive) {
+      assertTrue(queryExecutor.useBinaryForReceive(oid));
+    }
+    for (int oid : oidsSend) {
+      assertTrue(queryExecutor.useBinaryForSend(oid));
+    }
+    TestUtil.closeDB(con);
+  }
+
+  /**
+   * Make sure custom binary types are handled automatically.
+   *
+   * @throws SQLException if a database error occurs
+   */
+  @Test
+  public void testCustomBinaryTypes() throws SQLException {
+    con = TestUtil.openDB();
+
+    PgConnection pgconn = (PgConnection) con;
+    QueryExecutor queryExecutor = pgconn.getQueryExecutor();
```

Comment:
Do you think you could "disable default binary transfer for POINT" via existing connection properties rather than using a private `PgConnection + QueryExecutor` APIs?

Reply:
This might be possible, but as stated above I need to use the internal API anyway.
So it is more flat/simple to just remove this single type without introducing possible side effects by setting additional connection properties.